pallets/automation-time/src/lib.rs

```diff
@@ -1014,6 +1043,44 @@ pub mod pallet {
 			(task, <T as Config>::WeightInfo::run_auto_compound_delegated_stake_task())
 		}
 
+		/// Attempt to decode and run the call.
+		pub fn run_dynamic_dispatch_action(
+			caller: AccountOf<T>,
+			encoded_call: Vec<u8>,
+			task_id: TaskId<T>,
+		) -> Weight {
+			match <T as Config>::Call::decode(&mut &*encoded_call) {
+				Ok(scheduled_call) => {
+					let dispatch_origin: T::Origin =
+						frame_system::RawOrigin::Signed(caller.clone()).into();
+
+					let call_weight = scheduled_call.get_dispatch_info().weight;
+					let (maybe_actual_call_weight, result) =
+						match scheduled_call.dispatch(dispatch_origin) {
+							Ok(post_info) => (post_info.actual_weight, Ok(())),
+							Err(error_and_info) =>
+								(error_and_info.post_info.actual_weight, Err(error_and_info.error)),
+						};
+
+					Self::deposit_event(Event::DynamicDispatchResult {
+						who: caller,
+						task_id,
+						result,
+					});
+
+					maybe_actual_call_weight
+						.unwrap_or(call_weight)
+						.saturating_add(<T as Config>::WeightInfo::deposit_event_weigher())
```

Comment:
since there is no support for params with default values I would like to avoid that, but I could add `run_dynamic_dispatch_action_fail_decode` instead of `deposit_event_weigher`. `run_dynamic_dispatch_action_fail_decode` just includes a failed decode and an event. which should be roughly equivalent. Thoughts?

Reply:
That works for me. Your suggestion addresses my concern that any future developers who work on this method have to remember to benchmark any code they write here independently.

Reply:
The other option would be to use the system.remark as dispatch and subtract the weight of depositing the duplicate event. No preference from me.